table/tables/mutation_checker.go

```diff
@@ -47,20 +52,14 @@ type columnMaps struct {
 // Namely, assume the database is consistent before, applying the mutations shouldn't break the consistency.
 // It aims at reducing bugs that will corrupt data, and preventing mistakes from spreading if possible.
 //
+// 3 conditions are checked:
+// (1) row.value is consistent with input data
+// (2) the handle is consistent in row and index insertions
+// (3) the keys of the indices are consistent with the values of rows
+//
 // The check doesn't work and just returns nil when:
 // (1) the table is partitioned
 // (2) new collation is enabled and restored data is needed
-//
-// How it works:
-//
-// Assume the set of row values changes from V1 to V2, we check
-// (1) V2 - V1 = {added indices}
-// (2) V1 - V2 = {deleted indices}
-//
-// To check (1), we need
-// (a) {added indices} is a subset of {needed indices} => each index mutation is consistent with the input/row key/value
-// (b) {needed indices} is a subset of {added indices}. The check process would be exactly the same with how we generate
-// 		the mutations, thus ignored.
 func CheckIndexConsistency(
```

Comment:
For the `CheckIndexConsistency` and `checkRowInsertionConsistency` we could add some comments explaining the theoretical performance overhead such as looping though array and allocating objects.